src/schemas/clinical-resolvers/clinicalData.ts

```diff
@@ -0,0 +1,80 @@
+import { getPaginatedClinicalData, ClinicalDataVariables } from '../../clinical/clinical-service';
+import { ClinicalEntityData, ClinicalInfo } from '../../clinical/clinical-entities';
+import { ClinicalErrorsResponseRecord } from '../../common-model/entities';
+import { errorResolver } from './clinicalErrors';
+
+export type ClinicalEntityGQLData = {
+  programShortName: string;
+  clinicalEntities: ClinicalEntityDisplayData[];
+};
+
+// FE Clinical Data Query Response Payload
+export type ClinicalEntityDataResponse = ClinicalEntityGQLData & {
+  clinicalErrors?: ClinicalErrorsResponseRecord[];
+};
+
+// GQL Formatting
+type EntityDisplayRecord = { name: string; value: string };
+
+interface ClinicalEntityDisplayData extends Omit<ClinicalEntityData, 'records'> {
+  records: EntityDisplayRecord[][];
+}
+
+const convertClinicalDataToGql = (
+  programShortName: string,
+  clinicalEntities: ClinicalEntityData[],
+) => {
+  const clinicalDisplayData: ClinicalEntityDisplayData[] = clinicalEntities.map(
+    (entity: ClinicalEntityData) => {
+      const records: EntityDisplayRecord[][] = [];
+
+      entity.records.forEach((record: ClinicalInfo) => {
+        const displayRecords: EntityDisplayRecord[] = [];
+        for (const [key, val] of Object.entries(record)) {
+          const name = key === 'submitter_id' ? false : key;
```

Comment:
having `name` as either a `string | false` seems like a type nightmare waiting to happen?

Reply:
Fair, will fix

Reply:
OK updated, used a conditional `continue` to handle submitter_id
This original logic was written a long time ago, definitely worth an update